### ISSUE TITLE
`Bug` TFT upload fails when BLE scanner is enabled in ESPhome config

### ISSUE THREAD

**Issue by bugtoo** [Blackymas/NSPanel_HA_Blueprint#1744]
### TFT Version

4.2.4

### ESPHome Version

4.2.4

### Blueprint Version

4.2.4

### Panel Model

NSPanel EU Model

### What is the bug?

 TFT upload fails when BLE scanner is enabled in ESPhome config. Removing it from config completes the TFT upload without issues.

### Steps to Reproduce

1. Add BLE scanner to config
2. Upload a blank TFT file
3. Try to upload the full tft file: it will fail randomly between 1% and 60%, possibly due to BLE scan kicking in.

### Your panel's YAML

```yaml
substitutions:
  ###### CHANGE ME START ######
  device_name: "nsp17lavanderia" 
  wifi_ssid: "[redacted]"
  wifi_password: "[redacted]"

  nextion_update_url: [redacted]" # URL to local tft File

packages:
  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint
    ref: main
    files:
      - nspanel_esphome.yaml # Core package
      # - advanced/esphome/nspanel_esphome_advanced.yaml # activate advanced (legacy) elements - can be useful for troubleshooting
      # - advanced/esphome/nspanel_esphome_addon_climate_cool.yaml # activate for local climate (cooling) control
      # - advanced/esphome/nspanel_esphome_addon_climate_heat.yaml # activate for local climate (heater) control
    refresh: 1s

esp32:
  framework:
    type: esp-idf

esp32_ble_tracker:
  scan_parameters:
    active: True

# Set Wi-Fi power save mode to "LIGHT" as required for Bluetooth on ESP32
wifi:
  power_save_mode: LIGHT
```

**Comment by illuzn** [comment]
Duplicate of #1686 - upstream issue unrelated to this project.

**Comment by bugtoo** [comment]
@illuzn I am not sure it is a duplicate, since the panel is perfectly working and it is just the TFT upload that randomly stops because of (I think) the BLE scan starting in the middle of the data transfer. This issue happens after the panel has been properly flashed through serial interface and no partitioning gets in the middle, so I believe it is not related.

**Comment by bugtoo** [comment]
More context: the server is http (no tls) and local.

**Comment by edwardtfn** [comment]
@bugtoo, could you please add this to your yaml and try again?

```yaml
external_components:
  - source:
      type: git
      url: https://github.com/edwardtfn/esphome
      ref: nextion-17
    components: [ nextion ]
    refresh: 1s
```

This have a better memory management and logs memory usage when uploading TFT.
I would appreciate if you can share a log collected during your upload process.

**Comment by edwardtfn** [comment]
I'm closing this as it is quiet since a while. All the changes proposed are not part of the released version (v4.2.5) so I believe we don't have more to do here.
Please feel free to re-open this if the problem persists.
